### PR TITLE
[guide] [react] add note about deprecated lifecycles

### PR DESCRIPTION
**Pull request by meetzaveri**
There were still deprecated lifecycle being mentioned in two places. Thus by removing them, it will make clear sense of latest React API after 16.2.

### CODE REVIEW

**Review by ljharb — CHANGES_REQUESTED** [review]
These are not actually "deprecated" in the sense that they'll only be "unsafe with async rendering". Either way, we should leave these here, and let React itself (and eslint-plugin-react rules) warn on these methods as needed.

**Comment by meetzaveri** [comment]
This makes sense. But can we like address developer that this lifecycles are now no more in use like by adding a label in parenthesis or making seperate section for it ?

**Comment by ljharb** [comment]
I think a section for it that calls out that only in React 16.3+, certain methods should be avoided in favor of others, would be useful, yes.

**Comment by meetzaveri** [comment]
Okay. So should I make changes now in readme file acc. to it?

**Comment by ljharb** [comment]
Updating this PR to do that would be great!

**Comment by meetzaveri** [comment]
Sure. So should I finally add a section about avoiding lifecycles?

**Comment by ljharb** [comment]
Not avoiding; but using the ones appropriate for your React version. Not everyone is on the latest version.

File: react/README.md [hunk 1]
```diff
@@ -701,6 +702,15 @@ We don’t recommend using indexes for keys if the order of items may change.
   > Why? [`isMounted` is an anti-pattern][anti-pattern], is not available when using ES6 classes, and is on its way to being officially deprecated.
 
   [anti-pattern]: https://facebook.github.io/react/blog/2015/12/16/ismounted-antipattern.html
+  
+## Things to avoid
```
Comment (ljharb) [inline react/README.md]: This isn't the right phrasing here; someone using React 16.2 shouldn't be avoiding anything.

File: react/README.md [hunk 2]
```diff
@@ -701,6 +702,15 @@ We don’t recommend using indexes for keys if the order of items may change.
...
+
+Using deprecated APIs will create bad legacy code. Hence it's good to know that which APIs have been deprecated.
```
Comment (ljharb) [inline react/README.md]: All of these APIs are not "bad" or "legacy", they're solely not safe to use with async rendering. Can you rephrase this to have less of a judgemental connotation?

File: react/README.md [hunk 3]
```diff
@@ -701,6 +702,15 @@ We don’t recommend using indexes for keys if the order of items may change.
...
+
+These Component lifecycles are been deprecated since React version 16.3+ :
```
Comment (ljharb) [inline react/README.md]: Could we add, for each one, which methods should be used instead (with links to their documentation)?

**Review by ljharb — CHANGES_REQUESTED** [review]

**Comment by MDASI** [comment]
There were still deprecated lifecycle being mentioned in two places. so by removing them, it will make clear sense of latest React API after 16.2.
